File: server/cmd/api/chatconfig.go
package main

type ChatConfig struct {
	ChatRoomTTL                 uint64 `json:"chatRoomTTL"`
	MaxNumberOfParticipants     uint64 `json:"maxNumberOfParticipants"`
	MaxHistoryDurationInSeconds uint64 `json:"maxHistoryDurationInSeconds"`
	MaxMessagesLength           uint64 `json:"maxMessagesLength"`
}

File: webapp/src/ChatConfig.ts
import { z } from 'zod'

export const ChatConfigSchema = z.object({
  chatRoomTTL: z.number().positive(),
  maxNumberOfParticipants: z.number().positive(),
  maxHistoryDurationInSeconds: z.number().positive(),
  maxMessagesLength: z.number().positive(),
})

export type ChatConfig = z.infer<typeof ChatConfigSchema>
